firebase-firestore/ktx/src/test/kotlin/com/google/firebase/firestore/ktx/FirestoreUnitTestKtxListEncoder.kt

```diff
@@ -0,0 +1,83 @@
+package com.google.firebase.firestore.ktx
+
+import com.google.firebase.firestore.encoding.FirestoreAbstractDecoder
+import com.google.firebase.firestore.encoding.FirestoreAbstractEncoder
+import com.google.firebase.firestore.encoding.FirestoreSerializersModule
+import kotlinx.serialization.DeserializationStrategy
+import kotlinx.serialization.SerializationStrategy
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.AbstractDecoder
+import kotlinx.serialization.encoding.AbstractEncoder
+import kotlinx.serialization.encoding.CompositeDecoder
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.serializer
+
+/**
+ * An encoder that encodes a @[Serializable] object to a plain list of Firestore supported types. This encoder is only supposed to be used for test purpose.
+ */
+private class FirestoreUnitTestKtxListEncoder() : AbstractEncoder(),
+    FirestoreAbstractEncoder {
+
+    val list = mutableListOf<Any>()
+
+    override fun encodeFirestoreNativeDataType(value: Any) {
+        list.add(value)
```

Comment:
```suggestion
    override fun encodeFirestoreNativeDataType(value: Any) = list.add(value)
```

Reply:
`list.add(value)` 's return type is Boolean, I need to have a Unit return type, so I changed to `list.let { it.add(value) }`